tests/unit/array/mixins/test_eval_class.py

```diff
@@ -36,13 +36,120 @@ def test_eval_mixin_perfect_match(metric_fn, kwargs, storage, config, start_stor
     da1.embeddings = np.random.random([10, 256])
     da1_index = DocumentArray(da1, storage=storage, config=config)
     da1.match(da1_index, exclude_self=True)
-    r = da1.evaluate(da1, metric=metric_fn, strict=False, **kwargs)
+    r = da1.evaluate(ground_truth=da1, metric=metric_fn, strict=False, **kwargs)
     assert isinstance(r, float)
     assert r == 1.0
     for d in da1:
         assert d.evaluations[metric_fn].value == 1.0
 
 
+@pytest.mark.parametrize(
+    'storage, config',
+    [
+        ('memory', {}),
+        ('weaviate', {}),
+        ('sqlite', {}),
+        ('annlite', {'n_dim': 256}),
+        ('qdrant', {'n_dim': 256}),
+        ('elasticsearch', {'n_dim': 256}),
+        ('redis', {'n_dim': 256}),
+    ],
+)
+@pytest.mark.parametrize(
+    'metric_fn, kwargs',
+    [
+        ('r_precision', {}),
+        ('precision_at_k', {}),
+        ('hit_at_k', {}),
+        ('average_precision', {}),
+        ('reciprocal_rank', {}),
+        ('recall_at_k', {'max_rel': 9}),
+        ('f1_score_at_k', {'max_rel': 9}),
+        ('ndcg_at_k', {}),
+    ],
+)
+def test_eval_mixin_perfect_match_labeled(
+    metric_fn, kwargs, storage, config, start_storage
+):
+    da1 = DocumentArray.empty(10)
+    for d in da1:
+        d.tags = {'label': 'A'}
+    da1.embeddings = np.random.random([10, 256])
+    da1_index = DocumentArray(da1, storage=storage, config=config)
+    for d in da1_index:
+        d.tags = {'label': 'A'}
```

Comment:
is this still needed?

Reply:
No, you are right, I think I can remove it.